deploy/utils.py

```
@@ -270,6 +277,35 @@ async def set_redis_data(logger, redis_client, key, value, expiry=None):
         logger.error(f"Error storing data in Redis for key {key}: {e}")
 
 
+class Debouncer:
+    """Debounce-механізм для asyncio: виконує корутину лише після паузи без нових викликів."""
+
+    def __init__(self, delay: float):
+        self.delay = delay
+        self._task: asyncio.Task | None = None
+
+    async def call(self, coro_func):
+        """Скасовує попередній pending-виклик і планує новий через self.delay секунд."""
```

Ruff (0.15.2)

[warning] Docstring contains ambiguous `і` (CYRILLIC SMALL LETTER BYELORUSSIAN-UKRAINIAN I). Did you mean `i` (LATIN SMALL LETTER I)?

(RUF002)